### PR TITLE
Fix the LK+Windows and Catalyst+Lightning latest/latest failures 

### PR DESCRIPTION
**Pull request by maliasadi**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
- Remove `ControlledQubitUnitary` in favour of `C(QubitUnitary)` from the list of supported operations and the device toml file. The `stopping_condition` method guarantees the consistency of decompositions. This change also reduces the size of the generated IR and impact the performance of Catalyst for the computation of `C(QubitUnitrary)` as this controlled gate is currently decomposed to `QubitUnitary`, hardcoding the expanded controlled-unitary-matrix into the IR.
- Raise a clear error message when initializing `lightning.kokkos` with zero-qubit on Windows. Allocating 0-size Kokkos vectors not supported on Windows. Required to Pass LK+Windows CI. 
- Update toml files respectively to maintain support in Catalyst latest. Required to Pass Catalyst-Lightning latest/latest CI.

### CODE REVIEW

File: pennylane_lightning/lightning_kokkos/lightning_kokkos.toml [hunk 1]
```diff
@@ -58,8 +58,6 @@ StatePrep              = {}
 # Gates which should be translated to QubitUnitary
 [operators.gates.matrix]
 
-BlockEncode            = {properties = [ "controllable" ]}
```
Comment (dime10) [inline pennylane_lightning/lightning_kokkos/lightning_kokkos.toml:61]: I think the base version of BlockEncode still needs to be in here.

File: pennylane_lightning/lightning_qubit/lightning_qubit.toml [hunk 2]
```diff
@@ -2,22 +2,22 @@ schema = 2
...
-ControlledPhaseShift   = { properties = [ "invertible",                 "differentiable" ] }
-ControlledQubitUnitary = { properties = [ "invertible",                                  ] }
-CRot                   = { properties = [ "invertible"                                   ] }
-CRX                    = { properties = [ "invertible",                 "differentiable" ] }
-CRY                    = { properties = [ "invertible",                 "differentiable" ] }
-CRZ                    = { properties = [ "invertible",                 "differentiable" ] }
-CSWAP                  = { properties = [ "invertible",                 "differentiable" ] }
-CY                     = { properties = [ "invertible",                 "differentiable" ] }
-CZ                     = { properties = [ "invertible",                 "differentiable" ] }
+CNOT                   = { properties = [ "invertible", "controllable", "differentiable" ] }
```
Comment (dime10) [inline pennylane_lightning/lightning_qubit/lightning_qubit.toml:5]: Unfortunately non of the aliased control gates are controllable from a strict point on view, so the previous version is correct I think.

File: pennylane_lightning/core/src/simulators/lightning_kokkos/StateVectorKokkos.hpp [hunk 3]
```diff
@@ -112,6 +112,12 @@ class StateVectorKokkos final
             }
         }
 
+#ifdef _WIN32
+        PL_ABORT_IF_NOT(num_qubits,
+                        "LightningKokkos zero-qubit device initialization is "
+                        "not supported on Windows.");
+#endif
+
```
Comment (AmintorDusko) [inline pennylane_lightning/core/src/simulators/lightning_kokkos/StateVectorKokkos.hpp:120]: Does `default.qubit` support a zero-qubit state? I think that now that `default.qubit` does not need the number of qubits for initialization this is a question difficult to answer. But does it make sense for us to support a zero-size state vector? I mean, we could only abort for all cases.

Comment (maliasadi) [reply pennylane_lightning/core/src/simulators/lightning_kokkos/StateVectorKokkos.hpp:120]: yaeh, `default.qubit` supports the zero-qubit state similar to `lightning.qubit` and `lightning.kokkos` (on Linux and macOS). This feature is also required in Catalyst. The zero-qubit support of `lightning.kokkos` on Windows, however, is more of an issue with the initialization of Kokkos state-vector of size 1, capacity 2 for 0-qubit. I could get a workaround for this, but as this is not a high priority now, we can create a PR with a proper & clean fix later. I added this to the changelog as well as tests in this PR for future reference. Anyway, I'll open a separate WIP PR later today for further discussions.

File: pennylane_lightning/lightning_qubit/lightning_qubit.toml (hunk 2 above)
Comment (vincentmr) [inline pennylane_lightning/lightning_qubit/lightning_qubit.toml:5]: Just a matter of definition; there is no kernel for C(CNOT) but we can apply it as C(PauliX) moving a target wire into the control wires. Is it what we mean?

**Review by vincentmr — COMMENTED** [review]
Just a question to be clear.

File: pennylane_lightning/lightning_qubit/lightning_qubit.toml [hunk 4]
```diff
@@ -37,7 +37,7 @@ SingleExcitationPlus   = { properties = [ "invertible", "controllable", "differe
 SingleExcitation       = { properties = [ "invertible", "controllable", "differentiable" ] }
 S                      = { properties = [ "invertible", "controllable", "differentiable" ] }
 SWAP                   = { properties = [ "invertible", "controllable", "differentiable" ] }
-Toffoli                = { properties = [ "invertible",                 "differentiable" ] }
+Toffoli                = { properties = [ "invertible", "controllable", "differentiable" ] }
 T                      = { properties = [ "invertible", "controllable", "differentiable" ] }
 
 [operators.gates.decomp]
```
Comment (dime10) [inline pennylane_lightning/lightning_qubit/lightning_qubit.toml:43]: Line 56 needs to be updated as well:
```py
BlockEncode = {properties = [ "controllable" ]}
```

File: pennylane_lightning/lightning_qubit/lightning_qubit.toml (hunk 2 above)
Comment (maliasadi) [reply pennylane_lightning/lightning_qubit/lightning_qubit.toml]: haha.. just testing the patch discussing this morning... it's now in a separate branch. Removed them all!

Comment (maliasadi) [reply pennylane_lightning/lightning_qubit/lightning_qubit.toml]: yeah, it's related to my other branch removing the C(op) gates from `_operations` and it's outdated  ;) Removed them all here.

**Review by dime10 — APPROVED** [review]
Thanks @maliasadi!

File: tests/test_device.py [hunk 5]
```diff
@@ -65,6 +67,14 @@ def test_create_device_with_unsupported_mpi_buf_size():
...
+    (device_name == "lightning.kokkos" and sys.platform == "win32"),
+    reason="lightning.kokkos doesn't support 0 wires on Windows.",
+)
+@pytest.mark.skipif(
+    device_name == "lightning.gpu",
+    reason="lightning.gpu doesn't support 0 wires.",
+)
 @pytest.mark.skipif(
     device_name == "lightning.tensor",
     reason="lightning.tensor doesn't support 0 wires.",
```
Comment (AmintorDusko) [inline tests/test_device.py:80]:
```suggestion
@pytest.mark.skipif(
    device_name in ["lightning.gpu", "lightning.tensor"] ,
    reason=device_name+" doesn't support 0 wires.",
)

**Review by AmintorDusko — APPROVED** [review]
Nice work! I left a small suggestion, but not a blocker.

File: pennylane_lightning/lightning_qubit/lightning_qubit.toml [hunk 6]
```diff
@@ -27,7 +27,7 @@ PauliX                 = { properties = [ "invertible", "controllable", "differe
 PauliY                 = { properties = [ "invertible", "controllable", "differentiable" ] }
 PauliZ                 = { properties = [ "invertible", "controllable", "differentiable" ] }
 PhaseShift             = { properties = [ "invertible", "controllable", "differentiable" ] }
-QubitUnitary           = { properties = [ "invertible",                                  ] }
+QubitUnitary           = { properties = [ "invertible", "controllable", "differentiable" ] }
```
Comment (vincentmr) [inline pennylane_lightning/lightning_qubit/lightning_qubit.toml:30]: What is the meaning of `differentiable` again?

Comment (dime10) [reply pennylane_lightning/lightning_qubit/lightning_qubit.toml:30]: Supported in circuit with adjoint differentiation. Apparently that has been possible with QubitUnitary for a few months now?

Comment (vincentmr) [reply pennylane_lightning/lightning_qubit/lightning_qubit.toml:30]: Only if `QubitUnitary` is first decomposed into single-parameter gates (i.e. using ZXZ or ZYZ). And this is only possible for 1,2-qubit gates I think.

Comment (vincentmr) [reply pennylane_lightning/lightning_qubit/lightning_qubit.toml:30]: I mean, if you're using Lightning's adjoint classes.

Comment (maliasadi) [reply pennylane_lightning/lightning_qubit/lightning_qubit.toml:30]: @vincentmr Where does this logic exist in the src code?

Comment (vincentmr) [reply pennylane_lightning/lightning_qubit/lightning_qubit.toml:30]: Any differentiable gate is assumed to be of the form `e**(i*theta*G)` where `theta` is a scalar parameter and `G` is the so-called generator. Eventually one hits this [line](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/core/src/simulators/lightning_qubit/algorithms/AdjointJacobianLQubit.hpp#L309), where the state vector object `mu` tries to apply the generator of the gate. `QubitUnitary` does not have a generator, in Lightning or PennyLane, so the state-vector's `applyGenerator` should complain it cannot apply the generator. It is possible to decompose 1,2-qubit gates using ZYZ [decomposition](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/ops/op_math/decompositions/single_qubit_unitary.py) however.

**Review by vincentmr — APPROVED** [review]
Looks go to me now, thanks and cheers @maliasadi !

**Review by multiphaseCFD — APPROVED** [review]
Nice one! Thanks @maliasadi !

**Pull request merged**